ksql-cli/src/main/java/io/confluent/ksql/Ksql.java

```diff
@@ -67,6 +69,11 @@ public static void main(final String[] args) throws IOException {
       System.exit(-1);
     }
 
+    // ask for password if not set through command parameters
+    if (!options.getUserName().isEmpty() && !options.isPasswordSet()) {
+      options.setPassword(readPassword());
```

Comment:
Should we add a unit test for this functionality?

Reply:
I don't see any tests cases that verify the `ksql` command using parameters. The `KsqlTest` uses mocks and executes `ksql.run()` directly. 

I tried using unit tests, but there are a few static classes used in the main class. I could reorganize the code to set stubs, but how do I set stubs or mocks on static classes?

Do you know if there are tests that verify ksql commands are executed using the cli?

Reply:
> Do you know if there are tests that verify ksql commands are executed using the cli?

Have you looked at the ones in https://github.com/confluentinc/ksql/blob/master/ksql-cli/src/test/java/io/confluent/ksql/cli/CliTest.java? Not exactly what you're looking for but many come close.